.github/workflows/spread-tests.yaml

```diff
@@ -196,29 +236,35 @@ jobs:
           fi
 
           export SPREAD_USE_PREBUILT_SNAPD_SNAP=true
+          
+          if [ "${{ inputs.use-snapd-snap-from-master }}" = true ]; then
+            export SPREAD_USE_SNAPD_SNAP_URL=https://storage.googleapis.com/snapd-spread-tests/snapd-tests/snaps/snapd_master_amd64.snap
+          fi
 
-          if [[ "${{ inputs.systems }}" =~ amazon-linux-2023 ]]; then
-              # Amazon Linux 2023 has no xdelta, however we cannot disable
-              # xdelta on a per-target basis as it's used in the repack section
-              # of spread.yaml, which is shared by all targets, so all systems
-              # in this batch will not use delta for transferring project data
-              echo "Disabling xdelta support"
-              export NO_DELTA=1
+          # This could be the case when either there are not systems for a group or
+          # the list of tests to run is empty
+          if [ -z "$RUN_TESTS" ]; then
+            echo "No tests to run, exiting..."
+            exit 0
           fi
 
           # Add openstack backend definition to spread.yaml
           if [ "${{ inputs.backend }}" = openstack ]; then
               ./tests/lib/spread/add-backend tests/lib/spread/backend.openstack.yaml spread.yaml
           fi
 
-          # This could be the case when either there are not systems for a group or
-          # the list of tests to run is empty
-          if [ -z "$RUN_TESTS" ]; then
-            echo "No tests to run, exiting..."
-            exit 0
+          spread_list=$($SPREAD -list $RUN_TESTS 2>&1)
+
+          if [[ "$spread_list" =~ amazon-linux-2023 ]]; then
+              # Amazon Linux 2023 has no xdelta, however we cannot disable
+              # xdelta on a per-target basis as it's used in the repack section
+              # of spread.yaml, which is shared by all targets, so all systems
+              # in this batch will not use delta for transferring project data
+              echo "Disabling xdelta support"
+              export NO_DELTA=1
           fi
 
-          if "$SPREAD" -list $RUN_TESTS 2>&1 | grep -q "nothing matches provider filter"; then
+          if "$spread_list" | grep -q "nothing matches provider filter"; then
```

Comment:
Is this missing an echo? I don't get how this is supposed to work.

Reply:
You are definitely right. I'll fix it and then rerun the tests on snapcore to make sure there are no other bugs.